src/enzyme_ad/jax/Passes/PropagateConstantBound.cpp

```diff
@@ -263,7 +265,105 @@ struct PropagateConstantBoundsPass
           }
         }
       }
+
+      if (!tensor_types)
+        continue;
+
```

Comment:
can we separate the memref arg portion out of this PR (e.g. into https://github.com/EnzymeAD/Enzyme-JAX/pull/314), otherwise LGTM and go ahead and merge